git_greppers/processing.sh

```
@@ -0,0 +1,281 @@
+#!/bin/bash
+
+# Function to get detailed information about a merge commit
+get_merge_commit_info() {
+  local commit_hash=$1
+  local is_merge=false
+  local merge_info=""
+  
+  # Check if this is a merge commit (has more than one parent)
+  local parent_count=$(git cat-file -p "$commit_hash" | grep -c "^parent ")
```

Shellcheck (0.10.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)